### PR TITLE
Fixes #18800: fix invalid escape in parameter strings

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/18800

### CODE REVIEW

**Comment by gpoblon** [comment]
a part of this fix comes from an upper-level issue which should be resolved. Therefore this PR should not be reviewed until my next commit

**Comment by gpoblon** [comment]
PR updated with a new commit

File: rudder-lang/src/parser.rs [hunk 1]
```diff
@@ -379,6 +379,21 @@ fn enum_not_expression(i: PInput) -> PResult<PEnumExpressionPart> {
     )(i)
 }
 
+/// An unescaped string is a literal string delimited by '"""'.
```
Comment (peckpeck) [inline rudder-lang/src/parser.rs:382]: Why move this ? and is it just a move ?

**Review by peckpeck — APPROVED** [review]

**Comment by gpoblon** [comment]
OK, merging this PR

**Comment by gpoblon** [comment]
OK, merging this PR

**Comment by gpoblon** [comment]
OK, merging this PR

File: rudder-lang/src/technique.rs [hunk 2]
```diff
@@ -433,14 +433,14 @@ impl Parameter {
                 self.value
             )));
         }
-        Ok((format!("{:#?}", self.value), None))
+        Ok((format!("\"\"\"{}\"\"\"", self.value), None))
```
Comment (VinceMacBuche) [inline rudder-lang/src/technique.rs:436]: Here all parameters in rl files are unescaped strings, is that what we want ?? Don't we want to keep an unescaped (triple double quotes) string if we got one and and escaped (single double quote string if we got one (

**Pull request merged**